mlflow/java/client/src/main/java/org/mlflow/tracking/RunsPage.java

```diff
@@ -0,0 +1,91 @@
+package org.mlflow.tracking;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.mlflow.api.proto.Service.*;
+
+public class RunsPage implements Page<Run> {
+
+    private final String token;
+    private final List<Run> runs;
+
+    private final MlflowClient client;
+    private final List<String> experimentIds;
+    private final String searchFilter;
+    private final ViewType runViewType;
+    private final List<String> orderBy;
+    private final int maxResults;
+
+    /**
+     * Creates a fixed size page of Runs.
+     */
+    public RunsPage(List<Run> runs,
+                    String token,
+                    List<String> experimentIds,
+                    String searchFilter,
+                    ViewType runViewType,
+                    List<String> orderBy,
+                    int maxResults,
+                    MlflowClient client) {
+        this.runs = Collections.unmodifiableList(runs);
```

Comment:
❤️ immutability!

Reply:
Hm, I'm actually wondering why this is necessary. Presumably we're the only callers of this constructor right?